UnityProject/Assets/Scripts/Items/Implants/Organs/Heart.cs

```diff
@@ -11,20 +13,42 @@ public class Heart : BodyPartFunctionality
 
 		public bool CanTriggerHeartAttack = true;
 
+		public int heartAttackThreshold = -80;
+
 		public int SecondsOfRevivePulse = 30;
 
 		public int CurrentPulse = 0;
 
+		[ShowIf("isEMPVunerable")]
+		public int EMPResistance = 2;
+
 		private bool alarmedForInternalBleeding = false;
 
 		[SerializeField] private Reagent salt;
 
 		[SerializeField] private float dangerSaltLevel = 20f; //in u
 
+		public override void OnEmp(int strength)
+		{
+			if (isEMPVunerable == false) return;
+
+			if(DMMath.Prob(1/EMPResistance))
```

Comment:
This check is repeated per organ type. It could be placed in `BodyPartFunctionality.OnEmp` and then call a new virtual method if the check passes.